semantics/src/main/scala/aqua/semantics/expr/ConstantSem.scala

```diff
@@ -0,0 +1,39 @@
+package aqua.semantics.expr
+
+import aqua.model.{ConstantModel, Model}
+import aqua.parser.expr.ConstantExpr
+import aqua.semantics.Prog
+import aqua.semantics.rules.ValuesAlgebra
+import aqua.semantics.rules.names.NamesAlgebra
+import aqua.semantics.rules.types.TypesAlgebra
+import cats.free.Free
+import cats.syntax.functor._
+
+class ConstantSem[F[_]](val expr: ConstantExpr[F]) extends AnyVal {
+
+  def program[Alg[_]](implicit
+    V: ValuesAlgebra[F, Alg],
+    N: NamesAlgebra[F, Alg],
+    T: TypesAlgebra[F, Alg]
+  ): Prog[Alg, Model] = {
+    for {
+      defined <- N.constantDefined(expr.name)
+      t <- V.resolveType(expr.value)
+      model <- (defined, t, expr.mark) match {
+        case (Some(_), Some(_), true) =>
+          Free.pure[Alg, Model](
```

Comment:
Need to check that types conform: already defined value should match the latter expectations.